WordPress/gutenberg
# Disable drop cap control if text is aligned

## What?

Disables drop cap control for paragraph blocks if the text is aligned to:
- Center
- Left on RTL
- Right on LTR

Fixes https://github.com/WordPress/gutenberg/issues/11756

## Why?

The drop cap did not work if the text was aligned because the first letter was not aligned with the rest of the text.
The drop cap also did not work on RTL where the first letter was in the wrong position.

## How?

When the text is aligned:

- The drop cap control is disabled and displays a help text.
- The block attribute` dropCap` is updated as well as the .has-drop-cap CSS classname.

## Testing Instructions





**Please test the deprecation:**
1. Disable Gutenberg. Activate a theme that has drop cap enabled.
2. Enable a LTR language on the WordPress install.
3. Create a new post or page and add 4 paragraph blocks with the drop cap option enabled:
One each for no alignment, left, center and right alignment.
4. Apply the PR and enable Gutenberg. 
5. Refresh the post or page you just created, and confirm if there are any block validation errors.

**Testing the functionality:**
1. Repeat step 3.
2. Confirm that the drop cap option is available for the left aligned block and the block with no alignment.
3. Confirm that it is disabled for the centered and right alignment.
4. Switch to an RTL language.
5. Refresh the post or page and confirm that there are no block validation errors.

1. Please confirm that when you switch to RTL, the left aligned block still displays the drop cap, but the drop cap control is disabled for this alignment.
2. Open the Advanced panel in the block settings sidebar.
3. Confirm that the has-drop-cap class is in the CSS field. Remove it.
4. Confirm that you were able to remove the visual drop cap.

**Check the front:**
Confirm that the drop cap is placed on the right side of the RTL text and that the spacing (margin) is correct.


## Screenshots or screencast 
![Screen Shot 2022-07-14 at 08 20 58](https://user-images.githubusercontent.com/7422055/178914140-0c4fb5fe-24db-4481-9e90-07d122ae9381.png)
